modules/ezcontent_node/modules/ezcontent_smart_article/src/Plugin/Field/FieldWidget/SmartEntityReferenceAutocompleteTagsWidget.php

```diff
@@ -0,0 +1,156 @@
+    $ids = array_column($input['target_id'], 'target_id');
+    if (count($ids) !== count(array_flip($ids))) {
+      $form_state->setError($element, 'Field "' . $element['target_id']['#title'] . '" doesn\'t allow duplicates.');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element = parent::formElement($items, $delta, $element, $form, $form_state);
+    $element['target_id']['#tags'] = TRUE;
+    $element['target_id']['#default_value'] = $items->referencedEntities();
+    $entity = $items->getEntity();
+    $field_settings = $items->getFieldDefinition()->getSettings();
+    $image_field_name = $field_settings['image_fields'];
+    $uuid = $entity->uuid();
+    $element['#attached']['library'][] = 'ezcontent_smart_article/ezcontent_smart_article_libs';
+    $element['status_messages'] = [
+      '#type' => 'status_messages',
+      '#weight' => -10,
+    ];
+    // To handle form rebuild with.
+    if ($form_state->getFormObject()->getFormId() == 'entity_browser_smart_image_browser_form') {
+      $element['smart_tag_hidden'] = [
+        "#type" => 'textfield',
+        '#required' => TRUE,
+        '#prefix' => '<div class="hidden"  id="smart_tag_hidden-text-id">',
+        '#suffix' => '</div>',
+        '#weight' => 0,
+      ];
+    }
+    // Add generate tags button.
+    $element['auto_image_tags'] = [
+      '#prefix' => '<div class="image-tag-field-wrapper" id="auto-image-tags">',
+      '#suffix' => '</div>',
+      '#weight' => 0,
+    ];
+    $element['generate_image_tags'] = [
+      '#type' => 'submit',
+      '#value' => t('Generate Image Tags'),
+      '#weight' => 11,
+      '#data' => $uuid,
+      '#image_field_name' => $image_field_name,
+      '#prefix' => '<div>',
+      '#suffix' => '</div>',
+      '#ajax' => [
+        'callback' => [$this, 'ezcontentSmartImageTagsGenerateCallback'],
+        'wrapper' => 'auto-image-tags',
+        'effect' => 'fade',
+        'event' => 'click',
+        'progress' => [
+          'type' => 'throbber',
+          'message' => 'Please wait...',
+        ],
+      ],
+    ];
+    return $element;
+  }
+
+  /**
+   * Image Tags generator callback.
+   */
+  public function ezcontentSmartImageTagsGenerateCallback(array &$form, FormStateInterface $form_state) {
+    $uuid = $form_state->getTriggeringElement()['#data'];
+    $image_field_name = $form_state->getTriggeringElement()['#image_field_name'];
+    $media = $form_state->getValue([]);
+    $field_data = $media[$uuid][$image_field_name];
+    $fid = $field_data[0]['fids'][0];
+    if (empty($fid) && !empty($form_state->getValue($image_field_name))) {
+      $file_id = reset($form_state->getValue($image_field_name));
+      $fid = $file_id['fids'][0];
+    }
+    if (!empty($fid)) {
+      $file = File::load($fid);
```

Comment:
Try removing the file load, we should be able to fetch the object from form_state object.